darwin/future/core/datasets/list_datasets.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional
+
+from pydantic import parse_obj_as
+
+from darwin.future.core.client import Client
+from darwin.future.core.types import TeamSlug
+from darwin.future.core.types.common import QueryString
+from darwin.future.data_objects.dataset import Dataset, DatasetList
+
+
+def list_datasets(api_client: Client, team_slug: Optional[TeamSlug] = None) -> DatasetList:
+    """
+    Returns a list of datasets for the given team
+
+    Parameters
+    ----------
+    api_client : Client
+        The client to use to make the request
+    team_slug : Optional[TeamSlug]
+        The slug of the team to retrieve datasets for
+
+    Returns
+    -------
+    DatasetList
+
+    Raises
+    ------
+    HTTPError
+        Any errors that occurred while making the request
+    ValidationError
+        Any errors that occurred while parsing the response
+
+    """
+    response = (
+        api_client.get("/datasets", QueryString({"team": str(team_slug)})) if team_slug else api_client.get("/datasets")
```

Comment:
Is it even possible to get dataset for given team from out API? I'm pretty sure our backend ignores that and always returns datasets from team, that API_KEY is authorized for (one key = one team).

Reply:
So, I think that choice of team happens when creating api_client, and then we can ofc filter datasets by name etc (locally because that API doesn't have actual usable filters on its own).

Reply:
Will review.

Reply:
Have removed.